### PR TITLE
docs: Improve top-level README

### PR DESCRIPTION
**Pull request by jodh-intel**
Various improvements to the top-level README file:

- Moved the following sections from the runtime's README to the
  top-level README:
  - License
  - Platform support / Hardware requirements
- Added the following sections to the top-level README:
  - Configuration
  - Hypervisors
- Improved formatting of the Documentation section in the top-level
  README.
- Removed some unused named links from the top-level README.

Fixes: #3557.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: README.md
Comment (jodh-intel) [inline README.md:35]: @bpradipt - Can you provide some words for the PPC "Virtualization technology" entry? Should this row also mention "pSeries"?

File: README.md
Comment (jodh-intel) [inline README.md:36]: @Jakob-Naucke - please could you confirm that SIE is correct here? Should this row also mention "zSeries" or "Linux on Z" or whatever the latest term is? :)

File: README.md
Comment (jodh-intel) [inline README.md:80]: I'm undecided whether this is the best approach (reference the config details in the runtime's README) or whether we should just move those details to the top-level README since the config file doesn't just apply to the runtime. Thoughts folks?

**Comment by jodh-intel** [comment]
Ugh. The static checker is failing to validate a correct URL:

```
[static-checks.sh:584] ERROR: found HTTP error status codes for URL https://www.amd.com (403)
ERROR: Invalid URL 'https://www.amd.com' found in the following files:

README.md
```

The problem appears to be that the site in question is filtering on the user agent. Our CI checker doesn't set one, hence the HTTP 403 ("forbidden") error message.

For now, I've just removed that link since currently there is no good way to fix the static checker's URL check in a generic way.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

File: README.md
Comment (Jakob-Naucke) [reply README.md]: @jodh-intel thanks for the update. Yes, the official term is IBM Z & LinuxONE, and SIE would still be correct.

File: README.md
Comment (Jakob-Naucke) [inline README.md:145]: I think we could link straight to https://github.com/kata-containers/kata-containers/wiki/Glossary here while we're at it.

File: README.md
Comment (Jakob-Naucke) [reply README.md:80]: The config file doesn't just apply to the runtime, but doesn't everything that we describe in that README only apply to the runtime?

File: README.md
Comment (jodh-intel) [inline README.md:35]: "LinuxOne" has to be rendered in backticks to avoid failing the spell checker currently.

File: README.md
Comment (jodh-intel) [reply README.md:79]: @Jakob-Naucke - I'm not totally clear on what you're saying here. The license and platform support applies to the whole project so that's why I've moved it to the top-level. The config is a bit more debatable since although it's technically for the runtime (it's read by the runtime), it also affects other parts of the system (such as the agent). Are you saying we should move that config detail into the top-level too? That would make the information more visible but I left it in the runtime README since if we moved the full details to the top-level README, it felt a bit odd for the runtime's README to refer to another file for _it's_ configuration. It's debatable though ;)

File: README.md
Comment (jodh-intel) [inline README.md:34]: @Amulyam24 - could you talk at this line please?

File: README.md
Comment (Jakob-Naucke) [reply README.md:79]: Sorry, I'm too brief lately :) I meant: While the Kata Containers config does not only affect the runtime, AFAICT, everything we describe in https://github.com/kata-containers/kata-containers/tree/main/src/runtime#configuration does only affect the runtime. Thus, right now, I don't see a reason to move it up anyway.

Comment (jodh-intel) [reply README.md]: Ack. I've updated the Configuration for both README's to give a bit of extra information to users.

File: README.md
Comment (jodh-intel) [reply README.md:35]: https://github.com/kata-containers/tests/pull/4413 includes a fix for this.

File: README.md
Comment (Amulyam24) [reply README.md:34]: Hi @jodh-intel , this looks good to me.

File: README.md
Comment (jodh-intel) [reply README.md]: That PR has now landed so I've removed the backticks.

**Comment by jodh-intel** [comment]
Ping for reviews y'all!

File: README.md
Comment (Jakob-Naucke) [inline README.md:22]: Nit: I think "the Apache 2.0 license" is better, "an Apache 2.0 license" sounds a little as if we were using a custom variant. You're the native speaker though :)

File: src/runtime/README.md
Comment (Jakob-Naucke) [inline src/runtime/README.md:54]: What about the `/opt/kata` tree?

**Review by Jakob-Naucke — COMMENTED** [review]
Thank you @jodh-intel, also for the spelling updates. I have two other minor comments...

File: README.md
Comment (fidencio) [inline README.md:40]: I, personally, think that `kata-runtime check` is not the best way to check whether the host system is capable of running and creating a Kata Container.

As we're 100% Kubernetes focused nowadays, I'd rather see us working on and recommending [NFD](https://github.com/kubernetes-sigs/node-feature-discovery) instead.

But hey, I guess this is the best we have now.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jodh-intel!

File: README.md
Comment (egernst) [reply README.md:40]: I find the tool useful as a starting point, however -- most people will come here to test in their VM environment, and this helps. I do wonder if we need /want this here versus user/developer guide?

File: src/runtime/README.md
Comment (egernst) [reply src/runtime/README.md:54]: Yeah, there's a bit more detail here, indeed. it'd be more accurate to say what it will do by default. 

If we want to provide more detail, we can find a nice way to describe what actually happens, hopefully more succinctly than I do in my blather that follows:

It'll first check if a config path is specified in annotations associated with the pause container (yuck - I hope no one ever does this),  and then it'll check for a path specified within the CreateTaskRequest (this is what is done in the case of kata-deploy install, since we look at /opt/kata...), and then finally check the two default locations on the host filesystem, with /etc/kata-containers taking precedence over /usr/share/defaults/kata-containers....).  

See: https://github.com/kata-containers/kata-containers/blob/88a70d32ba5d3833ae84c8f65dd89fa17e3d0b1b/src/runtime/pkg/containerd-shim-v2/create.go#L212-L256

**Review by egernst — APPROVED** [review]
Nice improvements! I left a bit of feedback, mostly just agreeing with Jakob :)

**Review by egernst — COMMENTED** [review]
Good improvements, just some additional feedback.

File: README.md
Comment (jodh-intel) [reply README.md]: This text was dragged verbatim from the runtime README, but I agree so updated.

**Comment by jodh-intel** [comment]
@egernst, @Jakob-Naucke - ptal. The configuration is more confusing then even Eric listed. Hopefully, the verbiage on this PR will make things clearer :smile:

File: src/runtime/README.md
Comment (Jakob-Naucke) [inline src/runtime/README.md:51]: Nit: `its`

File: src/runtime/README.md
Comment (Jakob-Naucke) [inline src/runtime/README.md:55]: I think there shouldn't be a bullet for one element.

File: src/runtime/README.md
Comment (Jakob-Naucke) [inline src/runtime/README.md:68]: I see the merit of making patches cleaner by always having number 1. However, I think this makes it hard to read as plain text, which is a mission statement of Markdown IIRC.

**Review by Jakob-Naucke — COMMENTED** [review]
Thanks @jodh-intel -- I have some comments again

File: src/runtime/README.md
Comment (jodh-intel) [reply src/runtime/README.md]: Fixed.

File: src/runtime/README.md
Comment (jodh-intel) [reply src/runtime/README.md]: Fixed.

**Comment by jodh-intel** [comment]
Ping for re-review folks.

File: src/runtime/README.md
Comment (Jakob-Naucke) [inline src/runtime/README.md:134]: This requires containerd debug to be enabled. Should we mention that here?

Comment (jodh-intel) [reply src/runtime/README.md:134]: Fixed - ptal.

**Comment by jodh-intel** [comment]
@fidencio, @egernst - ptal so we can try to get this landed...

**Review by Jakob-Naucke — APPROVED** [review]

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jodh-intel!

**Review by GabyCT — APPROVED** [review]

**Pull request merged**
